lib/Transforms/Coroutines/CoroSplit.cpp

```diff
@@ -1549,8 +1549,10 @@ static void replacePrepare(CallInst *Prepare, CallGraph &CG) {
       for (auto &Use : Cast->uses()) {
         auto CS = CallSite(Use.getUser());
         if (!CS || !CS.isCallee(&Use)) continue;
-        PrepareUserNode->removeCallEdgeFor(CS);
-        PrepareUserNode->addCalledFunction(CS, FnNode);
+        PrepareUserNode->removeCallEdgeFor(
+            *cast<CallBase>(CS.getInstruction()));
+        PrepareUserNode->addCalledFunction(cast<CallBase>(CS.getInstruction()),
+                                           FnNode);
```

Comment:
Can you just `dyn_cast<CallBase>` and remove the use of `CallSite` completely?

Reply:
Just to make sure that I'm doing the right thing - just put this in a `if (auto *CB = dyn_cast<CallBase>(CS.getInstruction()))`?

Reply:
Go actually read the commit message for r358739.  Chandler is trying to remove `CallSite` completely; he just doesn't know about this use because it's still in our fork.

It should be easy to make this block of code stop using `CallSite` completely, assuming that `CallBase` provides an `isCallee` method.